packages/liquidation-sdk-viem/examples/whitelisted-erc4626-1inch.ts

```diff
@@ -240,41 +241,159 @@ export const check = async <
-                      throw Error(
-                        "could not fetch swap from both 1inch and paraswap",
-                      );
-
-                    dstAmount = BigInt(bestSwap.dstAmount);
-
-                    if (
-                      dstAmount < repaidAssets.wadMulDown(BigInt.WAD + slippage)
-                    )
-                      return;
-
-                    encoder
-                      .erc20Approve(srcToken, bestSwap.tx.to, srcAmount)
-                      .pushCall(
-                        bestSwap.tx.to,
-                        BigInt(bestSwap.tx.value),
-                        bestSwap.tx.data,
-                      );
-
+                    let retry = true;
+                    const tries: { srcAmount: bigint; srcToken: Address }[] =
+                      [];
+                    while (retry) {
+                      const bestSwap = await fetchBestSwap({
+                        chainId,
+                        src: srcToken,
+                        dst: market.params.loanToken,
+                        amount: srcAmount,
+                        from: executorAddress,
+                        slippage,
+                        includeTokensInfo: false,
+                        includeProtocols: false,
+                        includeGas: false,
+                        allowPartialFill: false,
+                        disableEstimate: true,
+                        usePermit2: false,
+                      });
+                      tries.push({ srcAmount, srcToken });
+
+                      if (!bestSwap)
+                        throw Error(
+                          "could not fetch swap from both 1inch and paraswap",
+                        );
+
+                      dstAmount = BigInt(bestSwap.dstAmount);
+
+                      if (
+                        dstAmount <
+                        repaidAssets.wadMulDown(BigInt.WAD + slippage)
+                      ) {
+                        // If we don't have enough liquidity, we try to swap to the alternative token and retry
+                        if (
+                          (srcToken === mainnetAddresses.usds ||
+                            srcToken === mainnetAddresses.dai ||
+                            srcToken === mainnetAddresses.sky ||
+                            srcToken === mainnetAddresses.mkr) &&
+                          chainId === ChainId.EthMainnet &&
+                          tries.length === 1
+                        ) {
+                          srcToken = Sky.getAlternativeToken(srcToken);
+                          retry = true;
+                        }
+                        // If even using the alternative token we still don't have enough liquidity, we try with both tokens (and half the amount)
+                        else if (
+                          Sky.isTokenPair(
+                            tries[0]?.srcToken,
+                            tries[1]?.srcToken,
+                          )
```

Comment:
If there are 2 tries (implied at this stage), it is necessarily a Sky token pair isn't it?

Reply:
In which case this simplifies to an `else` statement and the `isTokenPair` util is not useful anymore (we can get rid of it)

Reply:
yes but the previous `if` can also fail if the token is not a Sky related token

Reply:
Alright so we just need to check `tries.length == 2` right?

Reply:
Technically yes, as of now this is the only situation where we have 2 tries. I just thought it was "cleaner" to check it as well, but I don't mind removing it if you prefer